tests/lean/ll_infer_type_bug.lean

```diff
@@ -1,31 +0,0 @@
-
-partial def f : List Nat → Bool
-| []      => false
-| (a::as) => a > 0 && f as
-
-#print f._cstage2
```

Comment:
@leodemoura @zwarich FYI, because of how the old codegen adds these stage declarations to the environment via C++, ident elaboration will not be able to find them anymore after this PR. The old codegen itself is unaffected. I assume this is acceptable as the new codegen does not use the environment in this way anyway.